app/src/main/java/com/immichframe/immichframe/DimHttpServer.kt

```diff
@@ -0,0 +1,19 @@
+package com.immichframe.immichframe
+import fi.iki.elonen.NanoHTTPD
+
+class DimHttpServer(private val onDimCommand: (Boolean) -> Unit) : NanoHTTPD(8080) {
+
+    override fun serve(session: IHTTPSession): Response {
+        return when (session.uri) {
+            "/dim" -> {
+                onDimCommand(true)
+                newFixedLengthResponse("ImmichFrame dimmed")
+            }
+            "/undim" -> {
+                onDimCommand(false)
+                newFixedLengthResponse("ImmichFrame undimmed")
+            }
+            else -> newFixedLengthResponse(Response.Status.NOT_FOUND, MIME_PLAINTEXT, "Unknown command")
```

Comment:
We can start with it, I would like to try my wakelock approach as well. But why not support both? "dim" that maybe allows to show a clock, and a "turnoff" if the wakelock works as expected.

I'm not totally sure if it works without being a system app. But otherwise looks good for what it does.